coreapis/ldap/controller.py

```diff
@@ -133,19 +133,23 @@ def lookup_feideid(self, feideid, attributes):
 
     def health_check_thread(self, parent=None):
         while True:
-            servers = list(self.servers.values())
-            sleeptime = self.health_check_interval / len(servers)
-            for server in servers:
-                if parent and not parent.is_alive():
-                    return
-                if self.parse_ldap_config():
-                    break
-                server.check_connection()
-                time.sleep(sleeptime)
-            orgpools = self.orgpools.items()
-            for org, orgpool in orgpools:
-                self.host_statsd.gauge(self._org_statsd_key(org, 'alive_servers'),
-                                       len(orgpool.alive_servers()))
+            try:
+                servers = list(self.servers.values())
+                sleeptime = self.health_check_interval / len(servers)
+                for server in servers:
+                    if parent and not parent.is_alive():
+                        return
+                    if self.parse_ldap_config():
+                        break
+                    server.check_connection()
+                    time.sleep(sleeptime)
+                orgpools = self.orgpools.items()
+                for org, orgpool in orgpools:
+                    self.host_statsd.gauge(self._org_statsd_key(org, 'alive_servers'),
+                                           len(orgpool.alive_servers()))
+            except Exception as ex:
+                self.log.warn('Exception in health check thread', exception=str(ex))
```

Comment:
One thing to note here is that `str(ex)` normally doesn't include the class of the exception, nor where it came from, so if you are unlucky, you may get the error logged but still be unable to tell what it is or where it came from.